### PR TITLE
Support flags

### PR DESCRIPTION
**Pull request by brtkwr**

### CODE REVIEW

File: tasks/main.yml [hunk 1]
```diff
@@ -28,7 +28,13 @@
     --os-interface=public
     cluster-template-create
     {% for name, value in item.item.items() %}
+      {% if name == 'flags' %}
+        {% for flag in value %}
+    --{{ flag | replace('_', '-') }}
+        {% endfor %}
+      {% else %}
     --{{ name | replace('_', '-') }}='{{ value }}'
+      {% endif %}
```
Comment (markgoddard) [inline tasks/main.yml]: Any reason not to use an empty value to represent a flag?

Comment (brtkwr) [reply tasks/main.yml]: would that work?

Comment (brtkwr) [reply tasks/main.yml]: wouldn't that resolve to `--floating-ip-disabled=` which is a syntax error?

Comment (markgoddard) [reply tasks/main.yml]: Yes, you'd need to make a change to support it, e.g.

`{% if value is not none %}'{{ value }}'{% endif %}`

Comment (brtkwr) [reply tasks/main.yml]: oh. does this offer an advantage over iterating over a list of flags? would be more confusing for the reader to have empty variables no?

Comment (markgoddard) [reply tasks/main.yml]: I guess it's personal preference. Either way it needs documenting.

**Review by markgoddard — APPROVED** [review]
Do we need a change to p3-config for this? Jani suggested he'd seen the `flags:` syntax there.

**Comment by brtkwr** [comment]
Addressed in
- https://github.com/SKA-ScienceDataProcessor/p3-config/pull/54
- https://github.com/SKA-ScienceDataProcessor/p3-config/pull/55

**Pull request merged**